nixos/modules/services/display-managers/greetd.nix

```
@@ -108,6 +118,17 @@ in
         KeyringMode = "shared";
 
         Type = "idle";
+
+        StandardInput = lib.mkIf cfg.isTUI "tty";
+        StandardOutput = lib.mkIf cfg.isTUI "tty";
+        # Without this errors will spam on screen
+        StandardError = lib.mkIf cfg.isTUI "journal";
+
+        # Without these bootlogs will spam on screen
+        TTYPath = "/dev/tty1";
+        TTYReset = true;
+        TTYVHangup = true;
+        TTYVTDisallocate = true;
       };
```

```suggestion
      } // (lib.optionalAttrs cfg.isTUI {
        StandardInput = "tty";
        StandardOutput = "tty";
        # Without this errors will spam on screen
        StandardError = "journal";

        # Without these bootlogs will spam on screen
        TTYPath = "/dev/tty1";
        TTYReset = true;
        TTYVHangup = true;
        TTYVTDisallocate = true;
      });
```

I think it would be better not to touch service for graphical greeters at all.